src/routes/v2/slp.ts

```diff
@@ -331,10 +340,20 @@ async function listSingleToken(
       })
     }
 
+    //console.log(`formattedTokens: ${JSON.stringify(formattedTokens,null,2)}`)
+
     let t
     formattedTokens.forEach((token: any) => {
       if (token.id === req.params.tokenId) t = token
     })
+
+    // If token could not be found.
+    if(t === undefined) {
+      t = {
+        id: 'not found'
+      }
+    }
+
```

Comment:
Potentially breaking change here. When token couldn't be found, such as when a mainnet tx was passed to testnet or vice versa, this function would return `undefined`.

This code catches that use case and returns some intelligible data.